File: 1103-distribute-candies-to-people/1103-distribute-candies-to-people.java
class Solution {
    public int[] distributeCandies(int candies, int num_people) {
        int arr[] = new int[num_people];
        int i=0,p=1;
        while(candies>0){
            arr[i++] += candies > p?p:candies;
            candies-=p;
            p++;
            if(i==num_people){ 
                 i=0;
            }
        }
    return arr;
    }
}

File: 1103-distribute-candies-to-people/1103-distribute-candies-to-people.py
class Solution(object):
    def distributeCandies(self, candies, num_people):
        index=0
        give=1
        arr=[0] * num_people
        while candies > 0:
            arr[index%num_people] += min(give,candies)
            candies-=give
            index+=1
            give+=1
        return arr
        